Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -827,10 +827,27 @@ - (void)defaultTokenWithRetry:(BOOL)retry handler:(nullable FIRInstanceIDTokenHa
       if (!APNSRemainedSameDuringFetch && hasFirebaseMessaging) {
         // APNs value did change mid-fetch, so the token should be re-fetched with the current APNs
         // value.
-        [self retryGetDefaultTokenAfter:0];
+        [self.installations installationIDWithCompletion:^(NSString *_Nullable identifier,
```

Comment:
optional: May be worth refactoring to a separate method to improve readability.

Reply:
Done.